src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,106 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import classnames from 'classnames';
+import { Todo } from '../../react-app-env';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+interface Props {
+  todos: Todo[];
+  onSelectedUserId: (number: number) => void;
+}
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = ({ todos, onSelectedUserId }) => {
+  const [query, setQuery] = useState('');
+  const [selectedActivities, setSelectedActivities] = useState('');
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+  const filterTodos = useMemo(() => (
+    todos.filter(todo => {
+      const todoTitleToLowerCase = todo.title.toLowerCase();
+      const queryToLowerCase = query.toLowerCase();
+
+      return todoTitleToLowerCase.includes(queryToLowerCase);
+    })
+      .filter(todo => {
+        switch (selectedActivities) {
+          case 'active':
+            return !todo.completed;
+          case 'completed':
+            return todo.completed;
+          case 'select all':
+            return todo;
+          default:
+            return todo;
+        }
+      })
+  ), [query, todos, selectedActivities]);
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+  return (
+    <div className="TodoList">
+      <h2>
+        Todos:
+      </h2>
 
-          <button
-            className="TodoList__user-button button"
-            type="button"
+      <div className="TodoList__list-container">
+        <input
+          type="text"
+          placeholder="search"
+          value={query}
+          onChange={(event) => setQuery(event.target.value)}
+        />
+        <select
+          onChange={
+            (event) => setSelectedActivities(event.target.value)
+          }
+        >
+          <option
+            value="0"
+            disabled
+            selected
           >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+            select activities
+          </option>
+          <option>
+            active
+          </option>
+          <option>
+            completed
+          </option>
+          <option>
+            select all
+          </option>
```

Comment:
Simplify your own code, read [this](https://stackoverflow.com/questions/55601342/using-enumerations-in-react-select)